File: frontend/lib/api/preview.ts
import { apiClient } from "@/lib/api/client";

export type PreviewSimulatePayload = {
  prompt: string;
};

/** Calls backend preview/simulate. Stub returns JSON null until implemented. */
export async function simulatePreview(payload: PreviewSimulatePayload): Promise<null> {
  return apiClient<null>({
    path: "/api/v1/agent/preview/simulate",
    method: "POST",
    body: JSON.stringify({ prompt: payload.prompt })
  });
}

File: backend/app/api/v1/routes/preview.py
from fastapi import APIRouter
from fastapi.responses import JSONResponse

from app.schemas.preview import PreviewSimulateRequest

router = APIRouter()


@router.post("/simulate")
def simulate_preview(_payload: PreviewSimulateRequest) -> JSONResponse:
    """Placeholder until preview/simulation is implemented. Response body is JSON null."""
    return JSONResponse(content=None)
